### DISCUSSION TITLE
How do I actually grant permissions to plugins?

### DISCUSSION THREAD

**Discussion by adwaymalhotra** [zellij-org/zellij#2754] [Q&A] [answered]
I'm experimenting with writing a configurable tab bar, and I have managed to frakenstein some code from the default tab-bar plugin. 

First of all, the default plugin doesn't need to request `ReadApplicationState`. Why is that?

Second, I (as a user) don't know how to actually grant the permission to the plugin. I see the text with grant permission X? (y/n) but no way to interact with it? With other plugins I guess it makes sense if the plugin launches in a separate window, but not something like a tab-bar.

If I open a new tab, then the plugin doesn't seem to require the permission, and is rendered. This seems like a bug? 

This is my default layout if that matters (mintabs is my plugin):
```
layout {
  default_tab_template {
    children
    pane size=1 borderless=true {
      plugin location="zellij:mintabs"
    }
  }
  tab name="btop" {
    pane command="btop"
  }
  tab name="dots"  {
    pane {
      command "fish"
      args "-C cd ~/dotfiles; nvim"
    }
  }
  tab
}
```
dots tab:  
![dots tab](https://github.com/zellij-org/zellij/assets/9569615/e02f1458-29f8-467f-8e19-31091f6cfe7c)

new tab:  
![new tab](https://github.com/zellij-org/zellij/assets/9569615/b3a2e26f-83f8-4880-a05b-d1655f627db8)

**Comment by adwaymalhotra** [comment]
Figured out a way:

1. launch zellij with the plugin enabled.
2. Run: `zellij action launch-or-focus-plugin zellij:mintabs`. This focuses the plugin on the first tab.
3. Press `y`
4. kill zellij session.
5. Start new zellij session. The plugin works well in this session (and all future sessions).

**Comment by imsnif** [comment, marked as answer]
Hey, the built-in plugins don't need permissions because it's assumed that if you installed the application you trust it.

I think the best way to go about this is to set the plugin as non-selectable only after you receive permissions.

**Reply by adwaymalhotra** [reply]
oh, I see. I'll try writing to listen to the RequestPermissionResult event and set_selectable(false) after that.

Thanks!
